server/src/main/java/org/opensearch/action/search/ShardSearchFailure.java

```diff
@@ -88,12 +90,12 @@ public ShardSearchFailure(Exception e, @Nullable SearchShardTarget shardTarget)
         super(
             shardTarget == null ? null : shardTarget.getFullyQualifiedIndexName(),
             shardTarget == null ? -1 : shardTarget.getShardId().getId(),
-            ExceptionsHelper.detailedMessage(e),
-            ExceptionsHelper.status(ExceptionsHelper.unwrapCause(e)),
-            ExceptionsHelper.unwrapCause(e)
+            BaseExceptionsHelper.detailedMessage(e),
```

Comment:
Just an idea, may be could introduce private constructor here, it seems like we are unwrapping the cause at least 3 times here.

Reply:
Great suggestion!